src/script/lua_api/l_util.cpp

```diff
@@ -497,6 +499,92 @@ int ModApiUtil::l_colorspec_to_colorstring(lua_State *L)
 	return 0;
 }
 
+// colorspec_to_bytes(colorspec)
+int ModApiUtil::l_colorspec_to_bytes(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	video::SColor color(0);
+	if (read_color(L, 1, &color)) {
+		u8 colorbytes[4] = {
+			(u8) color.getRed(),
+			(u8) color.getGreen(),
+			(u8) color.getBlue(),
+			(u8) color.getAlpha(),
+		};
+		lua_pushlstring(L, (const char*) colorbytes, 4);
+		return 1;
+	}
+
+	return 0;
+}
+
+#include "log.h"
+
+// encode_png(w, h, data, [level])
+int ModApiUtil::l_encode_png(lua_State *L)
+{
+	// Swap endianness of u32 'n' in place
+	#define PNGBS(n) { u32 tmp = n ; writeU32( (u8*) ( & n ), tmp ); }
+
+	NO_MAP_LOCK_REQUIRED;
+
+	u32 width = readParam<float>(L, 1);
+	u32 height = readParam<float>(L, 2);
+	size_t scanline_count = height;
+	size_t scanline_size = width * 4;
+
+	size_t size;
+	const char *data = luaL_checklstring(L, 3, &size);
+
+	s32 compression = readParam<float>(L, 4);
+
+	std::ostringstream file, header, zdata, scanlines;
```

Comment:
I don't know where it makes a difference but these should be created as `std::ostringstream foo(std::ios::binary);`